draft-ietf-webtrans-http2.md

```diff
@@ -251,368 +278,411 @@ unidirectional, indicated by the second least significant bit of the
 stream ID.
 
 Because WebTransport does not provide an acknowledgement mechanism for
-WebTransport frames, it relies on the underlying transport's in order delivery
+WebTransport capsules, it relies on the underlying transport's in order delivery
 to inform stream state transitions. Wherever QUIC relies on receiving an ack
 for a packet to transition between stream states, WebTransport performs that
 transition immediately.
 
-# WebTransport Frames
+# WebTransport Capsules
 
-WebTransport frames mirror their QUIC counterparts as closely as possible to
-enable maximal reuse of any applicable QUIC infrastructure by implementors.
+WebTransport capsules mirror their QUIC frame counterparts as closely as
+possible to enable maximal reuse of any applicable QUIC infrastructure by
+implementors.
 
-A WebTransport frame begins with a Frame Type and Frame Length which are
-followed by zero or more fields that are type-dependent.
+WebTransport capsules follow the Capsule Protocol format defined in {
+{Section 3.2 of HTTP-DATAGRAM}}, containing a Capsule Type and Capsule Length
+which are followed by zero or more bytes of Capsule Value that are
+type-dependent.
 
 ~~~
-Frame {
-  Frame Type (i),
-  Frame Length (i),
-  Type-Dependent Fields (..),
+Capsule {
+  Capsule Type (i),
+  Capsule Length (i),
+  Capsule Value (..),
 }
 ~~~
-{: #fig-wt_frame_header title="WebTransport Frame Format"}
+{: #fig-Capsule_Protocol_header title="Capsule Format"}
 
-The Frame Type field indicates the type of the frame, defining what
+The Capsule Type field indicates the type of the capsule, defining what
 type-dependent fields will be present.
 
-The Frame Length field indicates the length of the WebTransport frame, including
-all type-dependent fields and other information. It does not include the size
-of the Frame Type or Frame Length fields themselves.
-
-Both of these fields use a variable-length integer encoding (see {{Section 16 of
-!RFC9000}}), with one exception. To ensure simple and efficient
-implementations of frame parsing, the frame type and length MUST use the
-shortest possible encoding. For example, for the frame types defined in this
-document, this means a single-byte encoding, even though it is possible to
-encode these values as a two-, four-, or eight-byte variable-length integer.
+The Capsule Length field indicates the length of the WebTransport capsule,
+including all type-dependent fields and other information. It does not include
+the size of the Capsule Type or Capsule Length fields themselves.
 
-## WT_PADDING Frames {#WT_PADDING}
+## PADDING Capsule {#PADDING}
 
-*[WT_PADDING]: #
+*[PADDING]: #
 
-A WT_PADDING frame (type=0x00) has no semantic value. PADDING frames can be used
-to introduce additional data between other WebTransport frames and can also be
-used to provide protection against traffic analysis or for other reasons.
+A PADDING capsule is an HTTP capsule {{HTTP-DATAGRAM}} of type=0x00 and has no
```

Comment:
Which number did the datagram capsule end up taking? ... [0x00](https://www.ietf.org/archive/id/draft-ietf-masque-h3-datagram-09.html#name-the-datagram-capsule)

Reply:
Yep, I left `0x00` for all the ones we hadn't generated randomly/decided on yet, but I'd imagine we could make PADDING be `0x01` since it's fairly reusable/carries no additional semantics. (Note that I omitted the `WT_` prefix from it for exactly that reason.)